File: backend/src/main/java/ch/fhnw/labeling_tool/user_group/CheckedOccurrence.java
package ch.fhnw.labeling_tool.user_group;

public class CheckedOccurrence {
    public final Long id;
    public final Long userId;
    public final CheckedOccurrenceLabel label;
    public final OccurrenceMode mode;

    public CheckedOccurrence(Long id, Long userId, CheckedOccurrenceLabel label, OccurrenceMode mode) {
        this.id = id;
        this.userId = userId;
        this.label = label;
        this.mode = mode;
    }
}

File: frontend/src/app/components/check/check/checked-occurrence.ts
import {OccurrenceMode} from './check.component';

export enum CheckedOccurrenceLabel {SKIPPED = 'SKIPPED', CORRECT = 'CORRECT', WRONG = 'WRONG'}

export class CheckedOccurrence {
  id: number;
  userId: number;
  label: CheckedOccurrenceLabel;
  mode: OccurrenceMode;


  constructor(id: number, userId: number, label: CheckedOccurrenceLabel, mode: OccurrenceMode) {
    this.id = id;
    this.userId = userId;
    this.label = label;
    this.mode = mode;
  }
}

export interface Occurrence {
  id: number;
  text: string;
}
